lib/strong_migrations/adapters/abstract_adapter.rb

```diff
@@ -50,12 +50,32 @@ def target_version(target_version)
         target_version ||= StrongMigrations.target_version
         version =
           if target_version && StrongMigrations.developer_env?
-            target_version.to_s
+            if target_version.is_a?(Hash)
+              if ActiveRecord::VERSION::STRING.to_f < 6.0
+                raise "Multiple databases are not supported by this ActiveRecord version"
+              end
+
+              target_version = target_version.stringify_keys
+              target_version.fetch(db_config_name) do
+                raise "StrongMigrations.target_version is not configured for :#{db_config_name}"
+              end
+            else
+              target_version.to_s
+            end
           else
             yield
           end
+
         Gem::Version.new(version)
       end
+
+      def db_config_name
+        if ActiveRecord::VERSION::STRING.to_f < 6.1
+          connection.pool.spec.name
```

Comment:
This always returns `primary` when running migrations in Rails 6 apps, so this feature may need to be limited to Rails 6.1+

Reply:
Can you clarify (I am not sure I got it)? 
This if-else is used because there was an API change some time ago - https://github.com/rails/rails/pull/37253.

Reply:
If you create a new Rails 6.0 app, set up multiple databases, and call `connection.pool.spec.name` inside a migration, it always returns `primary` when you run `rails db:migrate` (I'm guessing this is due to how Rails 6.0 internally handles connection switching between migrations).

Reply:
Yes, indeed. Thank you. Pushed new changes.